### PR TITLE
Mobile friendly layout

### PR DESCRIPTION
**Pull request by ghost**
A mobile friendly version of the page. The page has a hamburger icon, which when clicked slides the menu from the left side of the page.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -83,10 +83,39 @@ oppiaGithubPages.config(['$routeProvider', '$locationProvider', function($routeP
...
+    document.getElementById("page-sidebar").style.webkitTransform = "translate(0px, 0px)";
+    document.getElementById("page-overlay").style.display = "block";
+
+    // Register click events for the menu items and the overlay
+    document.getElementById("sidebar-menulist").addEventListener("click", $scope.hideSideBar, false);
+    document.getElementById("page-overlay").addEventListener("click", $scope.hideSideBar, false);
+  }
+
+  $scope.hideSideBar = function() {
+     document.getElementById("page-sidebar").style.webkitTransform = "translate(-100%, 0)";
```
Comment (seanlip) [inline app.js]: nit: indentation

File: index.html [hunk 2]
```diff
@@ -7,53 +7,41 @@
     <meta http-equiv="X-UA-Compatible" content="chrome=1">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <meta name="description" content="Oppia : A tool for creating interactive tutors">
-
     <link rel="stylesheet" type="text/css" media="screen" href="static/css/stylesheet.css">
-
+    <link rel="shortcut icon" type="image/png" href="static/images/favicon.png" />
```
Comment (seanlip) [inline index.html]: nit: no need to self-close at the end; follow previous link tags

File: index.html [hunk 3]
```diff
@@ -7,53 +7,41 @@
...
-        <h2 id="project_tagline" ng-bind="$root.projectTag"></h2>
-      </header>
-    </div>
-
-    <div id="main_content_wrap" class="outer">
-      <div>
-      <!--A container to make a grid with a fixed column-->
-        <div class="container row" style="width: 900px;">
-          <div ng-include="'pages/sidebar.html'"></div>
+    <div id = "page-overlay"></div>
```
Comment (seanlip) [inline index.html]: nit: remoev spaces around '='

Casing should follow other ids for consistency (it looks like they use underscores)

File: static/css/stylesheet.css [hunk 4]
```diff
@@ -533,12 +544,92 @@ Small Device Styles
...
 /*reduce the size of the image for small screen devices*/
-
 @media screen and (max-width: 640px) {
   img {
-    width: 640px;
+    width: 100%;
+  }
+}
+
+/* SMARTPHONES POTRAIT AND LANDSCAPE */
```
Comment (seanlip) [inline static/css/stylesheet.css]: PORTRAIT

**Comment by seanlip** [comment]
Hi @amgowano, two general things:
1. Why is this a PR to oppia:master? It should be to oppia:develop.
2. The .DS_Store file should not be included.

Thanks!

File: app.js (hunk 1 above)
Comment (ghost) [inline app.js]: Done.

File: index.html (hunk 2 above)
Comment (ghost) [inline index.html]: Done.

File: index.html (hunk 3 above)
Comment (ghost) [inline index.html]: Changed page-overlay and page-sidebar to use underscore.

File: static/css/stylesheet.css (hunk 4 above)
Comment (ghost) [inline static/css/stylesheet.css]: Done.

**Comment by seanlip** [comment]
Oops, sorry -- I retract (1). I forgot this was a separate repo!

**Comment by seanlip** [comment]
Also, the new page looks nice! However, I played with it a bit and I think I might have found a few bugs.

I started with the page being full-window (in Chrome, on a desktop) and then narrowed it. Then I clicked on the hamburger menu and got this:

scrollbars1

There seem to be three extra scrollbars on the left. It would be nice not to have any of them (or at least not to have the two inner ones).

I then widened the screen again. This leads to a weird glitch in which the left menu items don't seem to come back out to where they should be:

expandagain

Are you able to reproduce these? They seem worth fixing.

Thanks!

**Comment by seanlip** [comment]
Hi @amgowano , @MAKOSCAFEE , just checking: any news on this?

**Comment by makoscafee** [comment]
Sure it was @amgowano  task but i guess he has been busy lately.. I will check up on him and see what we can do ASAP.

**Comment by ghost** [comment]
I could reproduce the resizing bug but not the extra scroll bars bug. I fixed the resizing bug but I am not pleased with the solution.

Cause of the bug: For mobile devices what I do is translate the side menu 100% to the left and then once the hamburger  is pressed I translate the menu 0px and using css3 rules this gives the effect of the menu sliding.
When the user resizes the window and simulates a mobile device, closing the menu causes a translation. Resizing the window again results into a case where there is spacing between the menu and the content since the menu is translated to the left.

Solution:
My solution involves having two different side menus. One for large devices and one for small devices and one for large devices. I am not really happy with it but it fixes the bug for now.

**Comment by ghost** [comment]
Also Sean, I just saw the weird scroll bar issue that you were talking about. I just forked the repository on my personal laptop and saw it. It didn't show up on my work laptop which was what I was using to develop. I suppose I was using a different chrome version. I will work on a fix for that as well.

File: index.html [hunk 5]
```diff
@@ -7,53 +7,47 @@
...
-      </header>
-    </div>
-
-    <div id="main_content_wrap" class="outer">
-      <div>
-      <!--A container to make a grid with a fixed column-->
-        <div class="container row" style="width: 900px;">
-          <div ng-include="'pages/sidebar.html'"></div>
+    <div id="page_overlay"></div>
+    <div class="outer">
```
Comment (seanlip) [inline index.html:20]: should this (and the following lines) be indented by 2? If so we might be missing a closing div tag somewhere (but I'm not sure)

File: pages/sidebar.html [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <!--SIDEBAR-->
-<div id="sidebar">
-  <ul class="primary">
+<div id="sidebar" ng-class="{sidebar_fixed:(scrollPosition >= 220 && windowHeight)}">
```
Comment (seanlip) [inline pages/sidebar.html]: nit: add single space after :

File: static/css/stylesheet.css [hunk 7]
```diff
@@ -477,19 +462,49 @@ Full-Width Styles
...
+  left: 0;
+  right: 0;
+  display: none;
+  background-color: black;
+  z-index: 11;
+  opacity: 0.8;
 }
 
+#sidebar > .primary::-webkit-scrollbar { 
+    display: none;
```
Comment (seanlip) [inline static/css/stylesheet.css]: nit: indentation should be 2 spaces

**Comment by seanlip** [comment]
In general this looks great. Thank you @amgowano !

I have a few small nits, but once you fix them we can merge it. I think this is definitely an improvement on the current site!

File: pages/sidebar.html (hunk 6 above)
Comment (ghost) [inline pages/sidebar.html]: Done.

File: static/css/stylesheet.css (hunk 7 above)
Comment (ghost) [inline static/css/stylesheet.css]: Done.

File: index.html (hunk 5 above)
Comment (ghost) [inline index.html:20]: This is actually fine. <div id="page_overlay"> and <div class="outer"> (line 19 and 20 in green) are replacing <div id="header_wrap" class="outer> from above (line 21 red). The deleted block above has the closing divs deleted below at (line 39 to line 41 red).

Comment (seanlip) [inline index.html:20]: Ah sorry. I think I missed the closing tag on the `</div>`. Thanks!

**Comment by seanlip** [comment]
Great, thanks! Merging.

**Pull request merged**